### DISCUSSION TITLE
scheduler bind type flag not working on aarch64 machine

### DISCUSSION THREAD

**Discussion by EricLin89** [rabbitmq/rabbitmq-server#4949] [Other]
I'm using RabbitMQ 3.8.21 and erlang 23.3. The -stbt flag is the default value (db),
When running on a aarch64 machine, the command:
`erlang:system_info(scheduler_bind_type). `
returns 
`unbound`
and
`erlang:system_info(scheduler_bindings).` 
shows that all the sheduler threads are unbound.
and 
`erlang:system_info(scheduler_online).`
also gives an unexpected wrong number. 
Changing -stbt explicitly to tnnps also gives the same result.

While those results are correct on x86 machine.

Is this issue due to erlang can't recognise some types of CPU?
How can I confirm it and how to solve this problem?

**Comment by michaelklishin** [comment]
It very well may be that Erlang 23 does not support scheduler binding on ARM. Maybe Erlang 25 does,
it has introduced JIT support on ARM 64, so maybe there were other changes.

You can resolve this by trying Erlang 25 or contributing scheduler binding support on ARM 64 to Erlang/OTP.

FYI, RabbitMQ 3.8 s [out of support](https://www.rabbitmq.com/versions.html).
